src/ossl/ecdh.rs

```diff
@@ -106,12 +111,18 @@ impl ECDHOperation {
         {
             return Err(CKR_MECHANISM_PARAM_INVALID)?;
         }
+        let shared =
+            if !params.pSharedData.is_null() || params.ulSharedDataLen > 0 {
```

Comment:
should we need to check the `params.ulSharedDataLen` here? With this test, I think it will crash when somebody will provide NULL pointer + non-zero length (which is obviously invalid, but we should not crash).